src/compile/mark/geoshape.ts

```diff
@@ -0,0 +1,37 @@
+import {UnitModel} from '../unit';
+import * as mixins from './mixins';
+
+import {Config} from '../../config';
+import {Field, field, FieldDef, isFieldDef} from '../../fielddef';
+import {GEOSHAPE} from '../../mark';
+import {GEOJSON} from '../../type';
+import {VgGeoShapeTransform, VgPostEncodingTransform} from '../../vega.schema';
+import {getMarkConfig} from '../common';
+import {MarkCompiler} from './base';
+
+export const geoshape: MarkCompiler = {
+  vgMark: 'shape',
+  encodeEntry: (model: UnitModel) => {
+    const {config} = model;
+    return {
+      ...mixins.color(model),
+      ...mixins.text(model, 'tooltip'),
+      ...mixins.nonPosition('opacity', model)
+    };
+  },
+  postEncodingTransform: (model: UnitModel): VgPostEncodingTransform[] => {
+    const {encoding, markDef} = model;
+    const shapeDef = encoding.shape;
+
+    if ((isFieldDef(shapeDef) && shapeDef.type === GEOJSON) || markDef.type === GEOSHAPE) {
```

Comment:
> markDef.type === GEOSHAPE

Can this function be called when mark type is not geoshape since this is in geoshape file?

Reply:
Restructure a bit to include warnings?
```
if (isFieldDef(shapeDef)) {
   if (shapeDef.type === GEOJSON) {
      ...
   } else {
      ... throw warning that field type is not compatible with geoshape
   }
} else {
   ...throw warning that value definition is not supported for `shape` of geoshape
} 
```

Reply:
This comment seems unaddressed?

Reply:
This is wontfix for me. I don’t see why your code is better, and mine works. Yours causes marks to not get a post encode transform that should (when they’re just a geoshape and not geojson).

Reply:
My code might be wrong, but the current code is confusing. 

1) I don't understand why you have to check for mark type here it's already in the geoshape file.  
It seems unnecessary.  If you really allow other mark type to call this function, then this function belongs to a wrong place in the codebase. 

2) You don't have to use exactly my code, but you should throw warnings for unsupported case.